axelor-production/src/main/java/com/axelor/apps/production/db/repo/ManufOrderManagementRepository.java

```diff
@@ -52,8 +56,19 @@ public ManufOrder copy(ManufOrder entity, boolean deep) {
         operationOrder.setToConsumeProdProductList(null);
         operationOrder.setConsumedStockMoveLineList(null);
         operationOrder.setDiffConsumeProdProductList(null);
+        operationOrder.setBarCode(null);
       }
     }
     return super.copy(entity, deep);
   }
+
+  @Override
+  public ManufOrder save(ManufOrder entity) {
+    for (OperationOrder operationOrder : entity.getOperationOrderList()) {
+      if (operationOrder.getBarCode() == null) {
+        operationOrderRepo.save(operationOrder);
```

Comment:
Create separate method in operationOrderService to generate barcode. Use that method in operation order save and from this save.